src/tasks/mixin/search_mixin.py

```
@@ -50,27 +50,36 @@ def strafe_search(
             for key in keys:
-                result = check_func()
-                if result:
-                    return result
+                self.move_keys(key, duration=duration)
                 count += 1
                 if start is not None and self.active_time() - start >= time_out:
+                    self.move_keys(opposite[key], duration=duration)  # 超时前归正回原位
                     return None
-                self.move_keys(key, duration=duration)
+                result = check_func()
+                # 无论命中与否，检测后立即校验截止时间：check_func 自身耗时可能跨过截止线
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)